garminworkouts/garmin/garminclient.py

```diff
@@ -83,6 +83,14 @@ def update_workout(self, workout_id, workout):
                                     headers=GarminClient._REQUIRED_HEADERS, json=workout)
         response.raise_for_status()
 
+    def schedule_workout(self, workout_id, date):
+        #new method to schedule a workout
+        assert self.session
+        json_data = {"date": date}
```

Comment:
I would keep garminclient as thin as possible (hard to test), could you move payload generation to the Workout?